### DISCUSSION TITLE
Self hosted runners

### DISCUSSION THREAD

**Discussion by SatheeshRajuParayil** [community/community#39320] [Actions] [answered]
Hi Team,
Will adding a self hosted runner into the GitHub actions Workflow file runs the tests/code from the GitHub repository in local machine where runner is installed?

**Comment by nicholasbergesen** [comment, marked as answer]
Hi @SatheeshRajuParayil,

This sounds correct. When targeting a using a self-hosted runner in a workflow job, all the steps defined in the job will execute on that runner on the machine that is hosting it. If the job checks out repository code and runs tests, this will be done within the runner.
